### ISSUE TITLE
use sdl and the effect of dpi is very strange

### ISSUE THREAD

**Issue by TTopoo** [pthom/imgui_bundle#90]
1. I set HELLOIMGUI_USE_SDL_OPENGL3 and IMGUI_BUNDLE_WITH_SDL ON
2. runnerParams.backendType = HelloImGui::BackendType::Sdl;
3. my easy code
```
#include "include.h"

using namespace std;

void Gui()
{
    ImGui::Text("Test text");
}

int main()
{
    HelloImGui::SimpleRunnerParams runnnerParams;
    runnnerParams.guiFunction = Gui;
    runnnerParams.windowSize = { 400, 300 };

    HelloImGui::RunnerParams runnerParams = runnnerParams.ToRunnerParams();
    runnerParams.imGuiWindowParams.configWindowsMoveFromTitleBarOnly = true;
    runnerParams.backendType = HelloImGui::BackendType::Sdl;

    ImmApp::AddOnsParams addOnsParams;
    //addOnsParams.withMarkdown = true;
    //addOnsParams.withImplot = true;
    ImmApp::Run(runnerParams, addOnsParams);
    return 0;
}
```
4. Scaling to 100% is normal
![image](https://user-images.githubusercontent.com/58133115/222900207-aeb209a5-49bb-48e8-8aa6-726a940e58e7.png)

5. Scaling to 125% is abnormal. I test img button in this situation, the mouse cannot normally click on the displayed button, there seems to be a little deviation. But I can't take a good screenshot of this situation because my screenshot software can't see the mouse pointer
![image](https://user-images.githubusercontent.com/58133115/222900359-904b863f-1422-4405-963b-ed3cb5232af9.png)
![image](https://user-images.githubusercontent.com/58133115/222900871-028aded4-dd4a-45b6-9147-76de1cf5eed8.png)

6.When I switch backendType to Glfw, any scaling is normal.So I don't know if I made a mistake in using sdl. I sincerely hope I can get your help, and I'm sorry for my English level

**Comment by pthom** [comment]
Hello,

I tested your code on my side with a scaling factor of 350% and it worked fine. I could not reproduce your problem, not with GLFW and not with SDL. 

However DPI scaling on windows is a bit tricky. SDL is not DPI aware by default whereas glfw is, which is probably the reason why it works better with glfw.

I'm sorry I have not a solution at this time, I will keep thinking about it.

The fact that the text disappeared is strange. It might happen if the the font uses too much memory on the graphics card; but I do not think this was the issue here.

**Comment by pthom** [comment]
Hi,

AnyBackendEventCallback works only for SDL.

For glfw, you could use `glfwSetCursorPosCallback()` and co.. However their api is not easy, and I do not recommend that you use it if you are not familiar with C apis. If you use them, remember that they return a pointer to the previous callback and that you shall call it if it is not null.

However, there is maybe an easier solution:

- ImGui::GetIO().MouseDown[] gives you access to the mouse buttons status 
- ImGui::GetIO().MousePos gives you access to the mouse position
- To get the the glfw window pointer, you can write:
```cpp
GLFWwindow* win = (GLFWwindow *)HelloImGui::GetRunnerParams()->backendPointers.glfwWindow
```
 - glfwGetWindowPos and glfwSetWindowPos enable to get the window position and then move it according to the mouse movement

**Comment by TTopoo** [comment]
I tried the glfwSetCursorPosCallback function, which is very useful. 
However, new problems arise. If I only initialize once in Gui (runnnerParams.guiFunction = Gui). Then the input of ImGui will be invalid. After I found the reason in the answer of the highest evaluation found in [this link](https://stackoverflow.com/questions/71680516/how-do-i-handle-mouse-events-in-general-in-imgui-with-glfw), I wrote this function in front of ImGui:: NewFrame(), so that both the glfw callback and ImGui can respond normally. 
So I want to know whether there is a suitable method to register the callback function like the AnyBackendEventCallback of sdl? Perhaps we can add a step in AbstractRunner:: Setup() to register such callback functions?
My question may have deviated from the issue, I'm sorry!😓
